forge/1.20.1-mapping/src/main/java/org/mtr/mapping/mapper/ClickableWidgetExtension.java

```diff
@@ -0,0 +1,76 @@
+package org.mtr.mapping.mapper;
+
+import net.minecraft.client.gui.GuiGraphics;
+import net.minecraft.client.gui.narration.NarrationElementOutput;
+import org.mtr.mapping.annotation.MappedMethod;
+import org.mtr.mapping.holder.ClickableWidgetAbstractMapping;
+import org.mtr.mapping.holder.MutableText;
+import org.mtr.mapping.holder.Text;
+
+public class ClickableWidgetExtension extends ClickableWidgetAbstractMapping {
+
+	@MappedMethod
+	public ClickableWidgetExtension(int x, int y, int width, int height) {
+		this(x, y, width, height, "");
+	}
+
+	@MappedMethod
+	public ClickableWidgetExtension(int x, int y, int width, int height, String message) {
+		this(x, y, width, height, TextHelper.literal(message));
+	}
+
+	@MappedMethod
+	public ClickableWidgetExtension(int x, int y, int width, int height, MutableText message) {
+		super(x, y, width, height, new Text(message.data));
+	}
+
+	@MappedMethod
+	public void render(GraphicsHolder graphicsHolder, int mouseX, int mouseY, float delta) {
+	}
+
+	@Deprecated
+	@Override
+	public final void renderWidget2(GuiGraphics guiGraphics, int mouseX, int mouseY, float delta) {
+		GraphicsHolder.createInstanceSafe(guiGraphics, graphicsHolder -> render(graphicsHolder, mouseX, mouseY, delta));
+	}
+
+	@MappedMethod
+	public boolean mouseScrolled3(double mouseX, double mouseY, double amount) {
```

Comment:
Please override mouseScrolled2 and call mouseScroll3, the mapped mouseScrolled3 method would not get called otherwise (Forge)